File: src/person.js
'use strict';

class Person {
    constructor () {
        this._email;
        this._firstName;
        this._lastName;
    }
    
    get email () {

    }

    set email (email) {
        if (email.split('@').length !== 2)
            throw new Error('Invalid email.');
    }

    get name () {
        var firstName = this._firstName || '';
        var lastName  = this._lastName  || '';
        var name      = (firstName + ' ' + lastName).trim()
        return (name ? name : 'Ukjent navn');
    }

    set firstName (firstName) {
        this._firstName = firstName;
    }

    set lastName (lastName) {
        this._lastName = lastName;
    }
}

module.exports = Person;

File: src/user/person.py
class EmailException(Exception):
    pass

class Person(object):

    def __init__(self, firstName = "", lastName = "", email = None):
        self._firstName = firstName
        self._lastName  = lastName
        self.email      = email

    @property
    def name(self):
        return (self._firstName + " " + self._lastName).strip()

    @property
    def email(self):
        return self._email 

    @email.setter
    def email(self, email):
        if (email is not None) and ('@' not in email):
            raise EmailException("Invalid email address.")
        self._email = email
